File: core/src/com/gmaniliapp/flappyhermes/state/GameStateManager.java
package com.gmaniliapp.flappyhermes.state;

import com.badlogic.gdx.graphics.g2d.SpriteBatch;

import java.util.Stack;

public class GameStateManager {

    private Stack<com.gmaniliapp.flappyhermes.state.State> states;

    public GameStateManager() {
        states = new Stack<>();
    }

    public void push(com.gmaniliapp.flappyhermes.state.State state) {
        states.push(state);
    }

    public void pop() {
        states.pop().dispose();
    }

    public void set(State state) {
        pop();
        push(state);
    }

    public void update(float deltaTime) {
        states.peek().update(deltaTime);
    }

    public void render(SpriteBatch spriteBatch) {
        states.peek().render(spriteBatch);
    }
}

File: core/src/com/gmaniliapp/flappyhermes/state/GameStateManager.kt
package com.gmaniliapp.flappyhermes.state

import com.badlogic.gdx.graphics.g2d.SpriteBatch
import java.util.*

class GameStateManager {

    private val states: Stack<State> = Stack()

    fun push(state: State) {
        states.push(state)
    }

    fun pop() {
        states.pop().dispose()
    }

    fun set(state: State) {
        pop()
        push(state)
    }

    fun update(deltaTime: Float) {
        states.peek().update(deltaTime)
    }

    fun render(spriteBatch: SpriteBatch) {
        states.peek().render(spriteBatch)
    }

}